Remove WebCore::StorageQuotaManager

szewai opened the pull request:
#### 043b7ceee0be126ccc7bfd2659e8490959a8bf9a
<pre>
Remove WebCore::StorageQuotaManager
<a href="https://bugs.webkit.org/show_bug.cgi?id=262199">https://bugs.webkit.org/show_bug.cgi?id=262199</a>
rdar://problem/116129143

Reviewed by Youenn Fablet.

WebCore::StorageQuotaManager was introduced to be used in both WebKitLegacy and WebKit code. However, WebKit now uses
WebKit::OriginQuotaManager, and WebKitLegacy does not actually have quota management for IndexedDB (any quota increase
will be approved, see InProcessIDBServer::quotaManager), so WebCore::StorageQuotaManager becomes unnecessary.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
(WebCore::IDBServer::IDBServer::IDBServer):
* Source/WebCore/Modules/indexeddb/server/IDBServer.h:
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/storage/StorageQuotaManager.cpp:
(WebCore::StorageQuotaManager::create): Deleted.
(WebCore::StorageQuotaManager::StorageQuotaManager): Deleted.
(WebCore::StorageQuotaManager::requestSpaceOnMainThread): Deleted.
(WebCore::StorageQuotaManager::requestSpaceOnBackgroundThread): Deleted.
(WebCore::StorageQuotaManager::tryGrantRequest): Deleted.
(WebCore::StorageQuotaManager::updateQuotaBasedOnUsage): Deleted.
(WebCore::StorageQuotaManager::resetQuotaUpdatedBasedOnUsageForTesting): Deleted.
(WebCore::StorageQuotaManager::resetQuotaForTesting): Deleted.
* Source/WebCore/storage/StorageQuotaManager.h:
(): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::requestStorageSpace): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.messages.in:
* Source/WebKitLegacy/Storage/InProcessIDBServer.cpp:
(InProcessIDBServer::InProcessIDBServer):
(InProcessIDBServer::quotaManager): Deleted.
(storageQuotaManagerSpaceRequester): Deleted.
* Source/WebKitLegacy/Storage/InProcessIDBServer.h:

Canonical link: <a href="https://commits.webkit.org/268877@main">https://commits.webkit.org/268877@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2405a31013c3b19a9811471da10d037e1e5f697d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20919 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21324 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21992 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22809 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19490 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/21156 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24566 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21504 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21141 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20907 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18165 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23665 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/18069 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/18979 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/25278 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19154 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19171 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23190 "Passed tests") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/19740 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16769 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18989 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5023 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/23312 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19561 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
```diff
@@ -743,14 +739,14 @@ void IDBServer::requestSpace(const ClientOrigin& origin, uint64_t taskSize, Comp
 
     // Release lock because space requesting could be blocked.
     m_lock.unlock();
     result = m_spaceRequester(origin, taskSize);
```
```suggestion
    bool result = m_spaceRequester(origin, taskSize);
```

youennf commented on Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
```diff
@@ -743,14 +739,14 @@ void IDBServer::requestSpace(const ClientOrigin& origin, uint64_t taskSize, Comp
     ASSERT(m_lock.isHeld());
 
-    StorageQuotaManager::Decision result = StorageQuotaManager::Decision::Deny;
+    bool result = false;
```
```suggestion
```

youennf commented on Source/WebKitLegacy/Storage/InProcessIDBServer.h:
```diff
@@ -30,15 +30,19 @@
 #include <WebCore/IDBServer.h>
 #include <wtf/RefCounted.h>
 #include <wtf/RefPtr.h>
+
```
Not needed

szewai committed:
Remove WebCore::StorageQuotaManager
https://bugs.webkit.org/show_bug.cgi?id=262199
rdar://problem/116129143

Reviewed by Youenn Fablet.

WebCore::StorageQuotaManager was introduced to be used in both WebKitLegacy and WebKit code. However, WebKit now uses
WebKit::OriginQuotaManager, and WebKitLegacy does not actually have quota management for IndexedDB (any quota increase
will be approved, see InProcessIDBServer::quotaManager), so WebCore::StorageQuotaManager becomes unnecessary.

* Source/WebCore/Headers.cmake:
* Source/WebCore/Modules/indexeddb/server/IDBServer.cpp:
(WebCore::IDBServer::IDBServer::IDBServer):
* Source/WebCore/Modules/indexeddb/server/IDBServer.h:
* Source/WebCore/Modules/indexeddb/server/UniqueIDBDatabase.cpp:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/storage/StorageQuotaManager.cpp:
(WebCore::StorageQuotaManager::create): Deleted.
(WebCore::StorageQuotaManager::StorageQuotaManager): Deleted.
(WebCore::StorageQuotaManager::requestSpaceOnMainThread): Deleted.
(WebCore::StorageQuotaManager::requestSpaceOnBackgroundThread): Deleted.
(WebCore::StorageQuotaManager::tryGrantRequest): Deleted.
(WebCore::StorageQuotaManager::updateQuotaBasedOnUsage): Deleted.
(WebCore::StorageQuotaManager::resetQuotaUpdatedBasedOnUsageForTesting): Deleted.
(WebCore::StorageQuotaManager::resetQuotaForTesting): Deleted.
* Source/WebCore/storage/StorageQuotaManager.h:
(): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.cpp:
(WebKit::NetworkProcess::requestStorageSpace): Deleted.
* Source/WebKit/NetworkProcess/NetworkProcess.h:
* Source/WebKit/UIProcess/Network/NetworkProcessProxy.messages.in:
* Source/WebKitLegacy/Storage/InProcessIDBServer.cpp:
(InProcessIDBServer::InProcessIDBServer):
(InProcessIDBServer::quotaManager): Deleted.
(storageQuotaManagerSpaceRequester): Deleted.
* Source/WebKitLegacy/Storage/InProcessIDBServer.h:

Canonical link: https://commits.webkit.org/268877@main
